Add fflogs absorbed event

eatmorehufu opened the pull request:
And do nothing with it.

eatmorehufu committed:
Add absorbed event

ayyaruq commented:
Did fflogs change this again? I recall this event being removed due to a rename or something in early Endwalker.

eatmorehufu commented:
It's there now, fwiw. Here's an example:

```
{
    "timestamp": 452966,
    "type": "absorbed",
    "sourceID": 2,
    "sourceIsFriendly": true,
    "targetID": 6,
    "targetIsFriendly": true,
    "ability": {
        "name": "Eukrasian Diagnosis",
        "guid": 1002607,
        "type": 8,
        "abilityIcon": "012000-012954.png"
    },
    "fight": 3,
    "attackerID": 10,
    "attackerIsFriendly": false,
    "amount": 11372,
    "extraAbility": {
        "name": "attack",
        "guid": 872,
        "type": 128,
        "abilityIcon": "000000-000101.png"
    },
    "targetResources": {
        "hitPoints": 86098,
        "maxHitPoints": 86098,
        "mp": 10000,
        "maxMP": 10000,
        "tp": 0,
        "maxTP": 0,
        "x": 10004,
        "y": 9532,
        "facing": -158,
        "absorb": 52
    }
}
```

AzariahAeroborn committed:
CI commit

AzariahAeroborn reviewed (changes requested):
Lint failure is due to not adding a sample `absorbed` event to the fakeEvents record in the tests for the FFLogs adapter.  You should be able to drop your sample in there and then update the snapshot for the test file (`yarn test --updateSnapshot`) for the new test event -- the snapshot should result in no event being adapted from the test absorbed event.

eatmorehufu committed:
Make AbsorbedEvent it's own event, update tests

Because it has a bunch of extra fields. This can always be rolled into HealEvent if needed later, but it's not used so... whatever.

Updated tests.

NOTE: the event has a "fight: number" field, that I'm assuming is on all FFlogs events and xiva doesn't care about it.

eatmorehufu committed:
Add the seemingly unnecessary "fight" field.

AzariahAeroborn committed:
CI commit
